### PR TITLE
Create Protocols passing authentication in cleartext (ASIM Network Se…

### PR DESCRIPTION
**Pull request by praveenthepro**
…ssion schema)

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @praveenthepro, any reason behind adding multiple connector id's

**Review by v-prasadboke — DISMISSED** [review]

**Comment by v-prasadboke** [comment]
#10015 #10031
mentioning common content PR

File: Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+id: 5dca6047-24ed-4eb7-b44e-ec7f1bf42621
+name: Detect Outbound LDAP Traffic(ASIM Network Session schema)
+description: |
+  'Malicious actors often abuse misconfigured LDAP servers or applications that use the LDAP servers in organizations. Outbound LDAP traffic should not be allowed outbound through your perimeter firewall. This search will help determine if you have any LDAP connections to IP addresses outside of private (RFC1918) address space.'
```
Comment (v-prasadboke) [inline Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).txt:4]: Hello @praveenthepro, shorten the description below 255 characters.

File: Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+id: 6fd69768-fdf1-4cfd-805f-b243be3f0c6d
+name: Remote Desktop Network Traffic(ASIM Network Session schema)
+description: |
+  'This hunting query looks for network traffic on TCP/3389, the default port used
```
Comment (v-prasadboke) [inline Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml:4]: Here too. Shorten the description below 255 characters.

File: Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).txt (hunk 1 above)
Comment (praveenthepro) [reply Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).txt]: reduced the  description below 255

**Comment by v-prasadboke** [comment]
Hello @praveenthepro, Please let me know if you have any pending updates/modification from your side so that I can investigate the PR.

**Comment by praveenthepro** [comment]
> Hello @praveenthepro, Please let me know if you have any pending updates/modification from your side so that I can investigate the PR.

@v-prasadboke  go ahead and start investigating the PR

**Review by rahul0216 — COMMENTED** [review]
@praveenthepro Few suggestions for Hunting queries,
1. Add entity mappings
2. For every entity mapping extend the KQL as entity_0_field = column_name
3. Add version property
Refer this hunting query for reference, https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Windows%20Security%20Events/Hunting%20Queries/CommandsexecutedbyWMIonnewhosts-potentialImpacket.yaml

**Comment by praveenthepro** [comment]
> @praveenthepro Few suggestions for Hunting queries,
> 
> 1. Add entity mappings
> 2. For every entity mapping extend the KQL as entity_0_field = column_name
> 3. Add version property
>    Refer this hunting query for reference, https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Windows%20Security%20Events/Hunting%20Queries/CommandsexecutedbyWMIonnewhosts-potentialImpacket.yaml

@v-prasadboke added the entity and version

**Comment by v-prasadboke** [comment]
Screenshot 2024-03-05 112758
 
Hello @praveenthepro, I see you have committed changes which were requested by Rahul. But these are incomplete commits. 
Rahul has mentioned to update KQL query with adding **entity_0_field = column_name** to it. 

I have attached a screenshot for your reference. 

Thanks, 
Prasad

**Comment by praveenthepro** [comment]
> Screenshot 2024-03-05 112758
> Hello @praveenthepro, I see you have committed changes which were requested by Rahul. But these are incomplete commits. Rahul has mentioned to update KQL query with adding **entity_0_field = column_name** to it.
> 
> I have attached a screenshot for your reference.
> 
> Thanks, Prasad

@v-prasadboke Thanks Prasad, Added the new entities as like on the screenshot,

File: Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  _Im_NetworkSession
+  | where TimeGenerated > ago(1d) and EventResult != "Failure"
+  | where ipv4_is_private(SrcIpAddr) and not(ipv4_is_private(DstIpAddr)) and SrcIpAddr != DstIpAddr
+  | where tostring(DstPortNumber) has_any ("389", "636")
+  | summarize Starttime= min(TimeGenerated),EndTime= max(TimeGenerated),Eventscount=sum(EventCount), EventVendors=make_set(EventVendor,10) by SrcIpAddr,DstIpAddr,DstPortNumber,NetworkProtocol,EventResult
+  | extend IP_0_SrcIpAddr = SrcIpAddr
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).yaml:25]: For IP, identifier is Address
https://learn.microsoft.com/en-us/azure/sentinel/entities-reference
So it should be, 
- identifier: Address

File: Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+relevantTechniques:
+  - T1071
+  - T1059
+query: |
+  _Im_NetworkSession
+  | where TimeGenerated > ago(1d) and EventResult != "Failure"
+  | where ipv4_is_private(SrcIpAddr) and not(ipv4_is_private(DstIpAddr)) and SrcIpAddr != DstIpAddr
+  | where tostring(DstPortNumber) has_any ("389", "636")
+  | summarize Starttime= min(TimeGenerated),EndTime= max(TimeGenerated),Eventscount=sum(EventCount), EventVendors=make_set(EventVendor,10) by SrcIpAddr,DstIpAddr,DstPortNumber,NetworkProtocol,EventResult
+  | extend IP_0_SrcIpAddr = SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Detect Outbound LDAP Traffic(ASIM Network Session schema).yaml:21]: It should be
extend IP_0_Address = SrcIpAddr

File: Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    SchemaVersion: 0.2.4
+requiredDataConnectors: []
+
+tactics:
+  - CommandAndControl
+relevantTechniques:
+  - T1071
+
+query: |
+  _Im_NetworkSession
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml:16]: Same as above, use inbuilt filter.

File: Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  // Summarize data by session parameters
+  | summarize Starttime= min(TimeGenerated),EndTime= max(TimeGenerated),Eventscount=sum(EventCount), EventVendors=make_set(EventVendor,10) by SrcIpAddr,DstIpAddr,DstPortNumber,NetworkProtocol,EventResult
+  | extend PortUsage = case(
+      DstPortNumber == 23, "Telnet",
+      DstPortNumber == 143, "IMAP",
+      DstPortNumber == 110, "POP3",
+      DstPortNumber == 21, "FTP",
+      "Other"
+  )
+  | extend IP_0_SrcIpAddr = SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml:32]: It should be
extend IP_0_Address = SrcIpAddr

File: Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Protocols passing authentication in cleartext (ASIM Network Session schema).yaml:36]: It should be, 
identifier: Address

File: Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  - Schema: ASimNetworkSessions
+    SchemaVersion: 0.2.4
+requiredDataConnectors: []
+tactics:
+  - Lateral Movement
+relevantTechniques:
+  - T1021
+  - T1021.001
+query: |
+  _Im_NetworkSession
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml:15]: Same as above, use inbuilt filter.

File: Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+id: 6fd69768-fdf1-4cfd-805f-b243be3f0c6d
+name: Remote Desktop Network Traffic(ASIM Network Session schema)
+description: |
+  'This hunting query looks for unusual remote desktop activity by monitoring TCP/3389 traffic. While RDP is common, focus on atypical connections to identify potential threats.'
+tags:
+  - Schema: ASimNetworkSessions
+    SchemaVersion: 0.2.4
+requiredDataConnectors: []
+tactics:
+  - Lateral Movement
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml:10]: "LateralMovement" , No spaces

File: Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  // Filter events from the last day that were not failures
+  | where TimeGenerated > ago(1d) and EventResult != "Failure"
+  // Filter events where the source IP is private, the destination IP is not private, and the source and destination IPs are not the same
+  | where ipv4_is_private(SrcIpAddr) and not(ipv4_is_private(DstIpAddr)) and SrcIpAddr != DstIpAddr
+  // Filter events where the destination port number is 3389 (commonly used for Microsoft Remote Desktop (RDP))
+  | where tostring(DstPortNumber) has_any ("3389")
+  // Summarize the data by source IP, destination IP, destination port number, network protocol, and event result
+  // For each group, calculate the start time, end time, event count, and a set of up to 10 event vendors
+  | summarize Starttime= min(TimeGenerated),EndTime= max(TimeGenerated),Eventscount=sum(EventCount), EventVendors=make_set(EventVendor,10) by SrcIpAddr,DstIpAddr,DstPortNumber,NetworkProtocol,EventResult
+  | extend IP_0_SrcIpAddr = SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml:25]: It should be,
extend IP_0_Address = SrcIpAddr

File: Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: SrcIpAddr
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Hunting Queries/Remote Desktop Network Traffic(ASIM Network Session schema).yaml:29]: It should be,
identifier: Address

**Review by rahul0216 — COMMENTED** [review]
@praveenthepro Please address the inline comment.

File: Solutions/Network Session Essentials/Analytic Rules/Anomaly in SMB Traffic(ASIM Network Session schema).yaml [hunk 12]
```diff
@@ -0,0 +1,53 @@
+id: 8717e498-7b5d-4e23-9e7c-fa4913dbfd79
+name: Anomaly in SMB Traffic(ASIM Network Session schema)
+description: |
+  'This detection detects abnormal SMB traffic, a file-sharing protocol. By calculating the average deviation of SMB connections over last 14 days, flagging sources exceeding 50 standard deviations.'
```
Comment (rahul0216) [inline Solutions/Network Session Essentials/Analytic Rules/Anomaly in SMB Traffic(ASIM Network Session schema).yaml:4]: "50 standard deviations.' Need to update this one also.

**Review by rahul0216 — COMMENTED** [review]
@praveenthepro There is one more inline comment.

**Review by rahul0216 — APPROVED** [review]
Changes look good.

**Comment by v-prasadboke** [comment]
Thank you Rahul for the approval.
Will get this solution packaged and review it once.

**Review by v-prasadboke — APPROVED** [review]

**Pull request merged**